### ISSUE TITLE
Multiple connections when socket.io server is behind nginx

### ISSUE THREAD

**Issue by pzaj2** [socketio/socket.io-client#1126]
*Note*: for support questions, please use one of these channels: [stackoverflow](http://stackoverflow.com/questions/tagged/socket.io) or [slack](https://socketio.slack.com)

### You want to:

* [x] report a *bug*
* [ ] request a *feature*

### Current behaviour
When node application running socket.io server is behind nginx proxy, it does create multiple connections (starting with polling, ending up with two - usually - WS connections) to the same endpoint (originating from the same io.connect(...) call). This only happens on one of my pages, which is JS-heavy. With other pages where I only use JS to create socket.io connection it works as expected (single connection).

### Steps to reproduce (if the current behaviour is a bug)
Not sure how to reproduce it. I'm unable to share the code with you.

### Expected behaviour
A single request for long polling and single WS connection should be established.

### Setup
- OS: Ubuntu 16.04 (client & server), Windows 7 client + Ubuntu 16.04 server
- browser: Chrome (latest stable), Firefox (latest stable)
- socket.io version: 2.0.1

### Other information (e.g. stacktraces, related issues, suggestions how to fix)
I have no idea how to fix this, perhaps it's not a bug, rather than expected behaviour and there's a way to surpress this behaviour?

My client code for connecting is just that:
```js
window.ws = window.ws || {};
    window.ws.host = location.host + '/';
    window.ws.endpoint = '/websocket';
    window.ws.secure = true;

 var ioSocket = io.connect(window.ws.host, {
        path: window.ws.endpoint,
        secure: window.ws.secure,
        reconnectionAttempts: 10,
    });

window.ws.socket = ioSocket;
```

This is the only place this code is called, later on I only subscribe to different events with
```js
 window.ws.socket.on('event', ...);
```

![image](https://user-images.githubusercontent.com/19946083/26841598-e96c24ca-4ae1-11e7-97a7-ce11dd30e393.png)
This is my output from Networking -> XHR in Chrome. In WS tab there are 2 WS connections with SID matching 3rd and 4th XHR request from this output. It's always the same, just SID and t are changing obviously.

**Comment by darrachequesne** [comment]
@pzaj2 silly question, but are you sure `io.connect()` only gets called once? Else, I don't see any particular reason why two connections would be created...

Could you enable debugging with  `localStorage.debug='socket.io-client*'`?

**Comment by pzaj2** [comment]
@darrachequesne 
Yeah, I thought it is called only once, because all the calls originated in the same place. But it kept me wondering why only this one page. Did check all JS specific for this page with no success. Now, debugging didn't help me directly here, but because of it I noticed not only the connection is being repeated, but all code within this JS file. Found out I had it included twice... silly me.

Thank you for your help, I actually gave up on this for a moment, but you reminded by about my issue and I managed to solve it.

Is there any way to delete the issue completely? It's quite embarrassing to be honest :-1:
